backend/src/controller/map.rs

```diff
@@ -9,9 +9,31 @@ use actix_web::{
 
 use crate::config::auth::user_info::UserInfo;
 use crate::config::data::AppDataInner;
-use crate::model::dto::{MapSearchParameters, PageParameters};
+use crate::model::dto::{HeatMapQueryParams, MapSearchParameters, PageParameters};
 use crate::{model::dto::NewMapDto, service};
 
+#[utoipa::path(
+    context_path = "/api/maps",
```

Comment:
Focus on getting the API right first, so that this PR gets useful for frontend devs. Heatmaps are specific to a map and layer (other layers, like tree layer or alternative plant layers would have different heatmaps), so I would put it to `/maps/<id>/layers/plants/heatmap`.

The selected time influences the suggestions too. I assume we send this by the parameters? @Bushuo

Reply:
@Bushuo Would you mind having a look? I personally feel like it is easier to just return a matrix/vector from the backend and then generating the heatmap in the frontend using a library. I found two after a quick search:
- https://www.npmjs.com/package/react-heatmap-grid (seems simple to use)
- https://www.npmjs.com/package/heatmapjs (looks quite nice but no idea if compatible with react)

I would then build the API according to your wishes (I don't care if its an array, matrix, contains objects with keys, etc.).
You can use the endpoint as it currently is. It will simply return dummy data (0.5 in one corner; 0 everywhere else). Documentation can be found in swagger.

Reply:
@markus2330 Yes, sending via parameters is good. I did it like this for the timeline use case.

@kitzbergerg I think an matrix is better than a image. I think, with an array the frontend can more easily adjust the scale and crop the invisible parts related to the viewport size than with an image.
Thank you for searching the libraries, but i don't know if they will help us.
I thought of directly overlaying the heat-map in a separate layer inside of konva. Does not seem too difficult to implement from scratch.

Reply:
We decided today that we want to stay with a picture in the format PNG (more efficient than a textual matrix). It should be the same size as the base layer image, so that the scaling of the base layer can be reused.

Reply:
@Bushuo I now changed it to return an image. The image itself is really small as it is not clear in the backend how large it should be. So I would suggest leaving it like this in the backend and then scaling in the frontend. Otherwise different screen sizes, resolutions etc might lead to problems (I have no idea how image scaling would work in the backend). Is this ok for you or do you have a solution about how to set the size in the backend?

@markus2330 How should the size of the map be set? I think we would need something like a polygon selector when creating the map so it actually works. For now I just create dummy values in the frontend that will be sent when creating a map.
Additionally how does when merging and deploying? As soon as this is merged all previously created maps will be invalid as the geometry has a non null constraint?

Reply:
> I think we would need something like a polygon selector when creating the map so it actually works.

Yes, this is the plan, unfortunately the polygon selector is not yet implemented.

> Additionally how does when merging and deploying?

I don't understand the question.

> As soon as this is merged all previously created maps will be invalid as the geometry has a non null constraint?

This is currently no problem, we can reset the databases of dev and www.

Reply:
If there isn't a polygon yet, we probably want/need a default polygon. I suggest we use a square 100.000x100.000 Pixel, i.e. 1000x1000m, i.e. 1ha. This is also the maximum size allowed. One pixel of the heat map should have 10x10 Pixel of the Konva map, i.e. 10x10cm rastering. Please also add this to docu.

Reply:
I thought the coordinates where in meters? 
So for example a planting has x and y coordinates. They have to be whole number as they are defined as i32, so if the matrix should be in 10x10cm should the plantings coordinates be in cm?

Reply:
1px of kanvas = 1cm, unfortunately only written in doc/guidelines/ui.md